data/json/effects_on_condition/npc_eocs/generic_npc_eocs.json

```
@@ -373,5 +373,28 @@
         "unique_ids": [ "GUARD1", "GUARD2", "GUARD3", "GUARD4", "GUARD5", "GUARD6", "GUARD7" ]
       }
     ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_DIVING_INTO_WATER_CHECK",
+    "global": false,
```

```suggestion
    "global": false,
```
Does this not apply to NPCs because they don't dive or some other reason?

---

You don't really have to dive, you just have to swim. I think it would then apply to NPCs.

---

Ok then unless I'm misunderstanding this chart.
<html>
<body>
<!--StartFragment-->
EOC type | global | run_for_npcs | Runs when | Runs for
-- | -- | -- | -- | --
RECURRING | false | false | character is active | avater and all active NPCs
RECURRING | true | false | always | avater only
RECURRING | false | true | Inavailable | -
RECURRING | true | true | always | avater and all active NPCs
SCENARIO_SPECIFIC | false | false | character is active | avater only
SCENARIO_SPECIFIC | true | false | character is active | avater only
SCENARIO_SPECIFIC | false | true | Inavailable | -
SCENARIO_SPECIFIC | true | true | character is active | avater and all active NPCs
Others | false | false | activated | targeted character
Others | true | false | activated | targeted character
Others | false | true | Inavailable | -
Others | true | true | activated | avater and all active NPCs

<!--EndFragment-->
</body>
</html>

Global:false as you have it above means that it won't run on NPCs and if I'm wrong it also still won't run on NPCs but there wouldn't be anything you could do about it.

---

The chart is here https://github.com/CleverRaven/Cataclysm-DDA/issues/68224#issuecomment-1773986686

---

I've changed it to be global and to run on NPCs.